### PR TITLE
Parse MITgcm config files to check they are correct namelists

### PR DESCRIPTION
**Pull request by aidanheerdegen**
Parse MITgcm config files to check they are correct namelists adding to list self.config_files.

Addresses https://github.com/marshallward/payu/issues/148.

### CODE REVIEW

File: payu/models/mitgcm.py [hunk 1]
```diff
@@ -37,9 +37,23 @@ def __init__(self, expt, name, config):
...
+        for fname in files:
+            nml_parser = f90nml.Parser()
+            nml_parser.comment_tokens += '#'
+
+            data_nml = nml_parser.read(fname)
+            if len(data_nml) > 0:
+                self.config_files.append(fname)
+            else:
+                print("Excluding {0} from list of configuration files: "\
+                      "assumed to be not a namelist file (or empty)".format(fname))
```
Comment (marshallward) [inline payu/models/mitgcm.py]: can probably ditch the slash, I think string concatenation is automatic inside of parentheses.

**Comment by marshallward** [comment]
Sorry I have zonked out on this issue, does this mean that the list of valid input files can be determined from the namelists?

**Comment by aidanheerdegen** [comment]
Oh, I dunno, was that a possibility? This is just parsing them and making sure they're valid namelist files. As you surmised, it will also discard valid namelist files with no content, but all they need is an empty group to be considered valid.

So this is considered valid:
```fortran
! Packages
 &PACKAGES
 &
```

File: payu/models/mitgcm.py (hunk 1 above)
Comment (aidanheerdegen) [reply payu/models/mitgcm.py]: Fixed in the PEP8 cleanup

**Comment by marshallward** [comment]
Ah right, I see, I see.  Yeah that seems OK to me.

**Comment by aidanheerdegen** [comment]
I have done some light testing, seems to work ok .. gad I need to get some tests in the repo. Feel free to merge

**Pull request merged**